### DISCUSSION TITLE
How to fix the polygon outlines of the MVTlayer?

### DISCUSSION THREAD

**Discussion by mlfeasy** [visgl/deck.gl#6715] [Q&A]
![image](https://user-images.githubusercontent.com/44999087/155957790-379bafff-cf1b-4702-b757-df46998ca3e8.png)

How to fix these lines? Help ，please?

```
const countryLayer = new MVTLayer({
    id: 'country',
    data: `http://127.0.0.1:8090/hub/spaces/Country/tile/web/{z}_{x}_{y}.mvt`,
    minZoom: 0,
    maxZoom: 23,
    lineWidthMinPixels: 2,
    getLineColor: [5, 10, 40],
    getFillColor: [15, 40, 80]
})
```

**Comment by Pessimistress** [comment]
The lines are likely boundaries of the tile from one of the MVT layers. You can find them by inspecting the features `layerName` property. You can then provide a function to `getLineColor` or `getLineWidth` and make those lines either transparent or width: 0.

**Reply by mlfeasy** [reply]
These lines become part of polygon elements, and it is difficult to set the width or color for them individually. And there may be problems in selecting and highlighting. The best solution seems to be to fuse these polygons.

**Reply by Pessimistress** [reply]
I'm not sure I follow. If you generated these tiles yourself, you should add a buffer around the tiles, see https://github.com/visgl/deck.gl/issues/5192#issuecomment-734876226

**Reply by mlfeasy** [reply]
Thanks for your help.
I try to add buffer to my MVT Layer.  

![image](https://user-images.githubusercontent.com/44999087/156495996-de95f3c8-ed0e-4d5d-8433-7631dc5552b2.png)

For simple line styles, it works.
But It does not work  when I use dash line  _(PathStyleExtension)_.
 
 This is my code:
```
       const airspaceLayer = new MVTLayer({
        id: "airspace",
        data: `http://192.168.125.71:8090/hub/spaces/airspace/tile/web/{z}_{x}_{y}.mvt`,
        minZoom: 0,
        maxZoom: 23,
        binary: false,
        renderSubLayers: (props) => {
          const { z } = props.tile;
          return [
            new GeoJsonLayer(props, {
              id: `airspace-${props.id}-dashline`,
              visible: true,
              lineWidthUnits: "pixels",
              getLineColor: [51, 153, 204],
              getLineWidth:  getSegmentWidth(z),
              lineWidthMaxPixels: 20,
              // lineCapRounded: true,
              lineJointRounded: true,
              filled: false,
              lineWidthMinPixels: 1,
              getDashArray: [1, 6],
              dashJustified: true,
              dashGapPickable: true,
              extensions: [new PathStyleExtension({ dash: true })],
            }),
            new GeoJsonLayer(props, {
              id: `airspace-${props.id}-simpleline`,
              visible: true,
              lineWidthUnits: "pixels",
              getLineColor: [51, 153, 204],
              getLineWidth: getSegmentWidth(z),
              lineWidthMaxPixels: 20,
              filled: false,
              // lineCapRounded: true,
              lineJointRounded: true,
              lineWidthMinPixels: 1,
            }),
          ];
        },
      });
```

And buffer seems not works in globe view.

I don't know if it's a bug or my mistake

![image](https://user-images.githubusercontent.com/44999087/156495210-a840e6d9-ea77-40c0-b275-b38c236f2a07.png)

**Reply by Pessimistress** [reply]
You should not overwrite the `extensions` prop in `renderSubLayers`. Move the PathStyleExtension to the MVTLayer.

**Reply by mlfeasy** [reply]
Oh, it works in Mapview, Thanks so much.
How about GlobeView? Adding buffers doesn't seem to work

**Reply by mlfeasy** [reply]
It looks Like this
[https://codepen.io/xiantianxu/pen/zYPyjNv](https://codepen.io/xiantianxu/pen/zYPyjNv)
